src/main/java/fm/last/moji/impl/FileUploadOutputStream.java

```diff
@@ -80,40 +81,62 @@ public void flush() throws IOException {
   @Override
   public void close() throws IOException {
     log.debug("Close called on {}", this);
-    long size = -1L;
+
     try {
+      // Step 1. finish data writing to store node
+      long size = -1L;
       try {
         delegate.flush();
         size = delegate.getByteCount();
+        log.debug("Bytes written: {}", size);
+
+        String message = httpConnection.getResponseMessage();
```

Comment:
You can move this to line 96 since we're now only outputting the message on error (or put the else statement back that prints this in debug for the "happy day" scenario).